### PR TITLE
Fix ctc codestyle

### PR DESCRIPTION
**Pull request by unnonouno**
I fixed code style of CTC

### CODE REVIEW

File: chainer/functions/loss/ctc.py [hunk 1]
```diff
@@ -149,41 +150,36 @@ def calc_trans(self, path, yseq, rr, xp):
...
 
         total_probability = _logsumexp(
-            forward_prob_trans[0] + backward_prob_trans[0], xp, axis=1)
+            self.forward_prob_trans[0] + self.backward_prob_trans[0],
+            xp, axis=1)
         scale = grad_output[0] / batch_size
-        for t in six.moves.range(len(yseq)):
-            multiply = forward_prob_trans[t] + backward_prob_trans[t]
-            y = yseq[t]
+        for t in six.moves.range(len(self.yseq)):
```
Comment (okuta) [inline chainer/functions/loss/ctc.py]: How about　`for t, y in enumarate(self.yseq):`?

Comment (unnonouno) [inline chainer/functions/loss/ctc.py]: How about `for y, f, b in zip(yseq, forward_prob_trans, backward_prob_trans):` ?

**Comment by okuta** [comment]
Would you like to be to return sum of `forward_prob_trans` and `backward_prob_trans` in `calc_trans`?

**Comment by okuta** [comment]
LGTM!

**Pull request merged**